cmake/vara_feature_macros.cmake

```diff
@@ -82,3 +82,48 @@ int main() { return 0; }
 " ${varname})
   set(CMAKE_REQUIRED_FLAGS ${OLD_CMAKE_REQUIRED_FLAGS})
 endfunction()
+
+function(check_std_filesystem_flags varname)
+  # Check if we need to add -lstdc++fs or -lc++fs or nothing
+  set(OLD_CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES})
+  set(OLD_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+  set(CMAKE_REQUIRED_FLAGS "-std=c++17 ${CMAKE_REQUIRED_FLAGS}")
+
+  CHECK_CXX_SOURCE_COMPILES("
+#include <filesystem>
+int main(int argc, char ** argv) {
+std::filesystem::path p(argv[0]);
+return p.string().length(); }
```

Comment:
Can you move the duplicated code into a local variable, so we share it between all three compile checks?

Reply:
I tried that with `set(TEST_PROG "...")` but then `CHECK_CXX_SOURCE_COMPILES(${TEST_PROG} TEST_RESULT)` would always give an error.

Reply:
But would it not be `CHECK_CXX_SOURCE_COMPILES("${TEST_PROG}" TEST_RESULT)`

I think this should work, but maybe I'm wrong an CHECK_CXX_SOURCE is special somehow?

Reply:
Hmm, the extra `""` could be the solution.